tests/main/document-portal-activation/task.yaml

```diff
@@ -17,75 +17,75 @@ description: |
 # slot, and Amazon Linux because it doesn't have the required Python 3
 # packages to run the test.
-systems: [ -ubuntu-core-*, -amazon-linux-2-*, -centos-* ]
-
-environment:
-    XDG_RUNTIME_DIR: /run/user/$(id -u)
+systems:
+    - -amazon-linux-2-*
+    - -centos-*
+    - -ubuntu-14.04-*
+    - -ubuntu-core-*
 
 prepare: |
-    #shellcheck source=tests/lib/pkgdb.sh
-    . "$TESTSLIB"/pkgdb.sh
     #shellcheck source=tests/lib/snaps.sh
     . "$TESTSLIB"/snaps.sh
     install_local test-snapd-desktop
     install_local test-snapd-sh
     rm -f /usr/share/dbus-1/services/fake-document-portal.service
-    mkdir -p "$XDG_RUNTIME_DIR"
-    rm -rf "${XDG_RUNTIME_DIR:?}/*" "${XDG_RUNTIME_DIR:?}/.[!.]*"
+    session-tool --prepare -u test
+
+    # The fake portal writes to current working directory so let's give it permission to.
+    touch doc-portal.log
+    touch report-error.txt
+    chown test doc-portal.log
+    chown test report-error.txt
 
 restore: |
-    #shellcheck source=tests/lib/pkgdb.sh
-    . "$TESTSLIB"/pkgdb.sh
-    [ -f dbus-launch.pid ] && kill "$(cat dbus-launch.pid)"
     rm -f /usr/share/dbus-1/services/fake-document-portal.service
-    rm -rf "${XDG_RUNTIME_DIR:?}/*" "${XDG_RUNTIME_DIR:?}/.[!.]*"
+    rm -f doc-portal.log
+    rm -f report-error.txt
+    session-tool --restore -u test
 
 execute: |
     #shellcheck source=tests/lib/systems.sh
     . "$TESTSLIB"/systems.sh
 
-    check_stderr() {
-        test -n "$1"
-        if is_cgroupv2; then
-            [ "$(cat "$1")" = "WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement" ]
-        else
-            [ "$(wc -c < "$1")" -eq 0 ]
-        fi
-    }
-    echo "No output on stderr when running without a session bus"
-    user_data="$HOME/snap/test-snapd-desktop/current"
-    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
-    check_stderr stderr.log
-
-    echo "Starting session bus"
-    eval "$(dbus-launch --sh-syntax)"
-    echo "$DBUS_SESSION_BUS_PID" > dbus-launch.pid
+    if is_cgroupv2; then
+        check_stderr() {
+            test -n "$1"
+            test "WARNING: cgroup v2 is not fully supported yet, proceeding with partial confinement" = "$(cat "$1")"
+        }
+    else
+        check_stderr() {
+            test -n "$1"
+            if [ -s "$1" ]; then
+                echo "stderr contains some messages"
+                cat "$1"
+                exit 1
+            fi
+        }
+    fi
 
     echo "No output on stderr when running with a session bus, when xdg-document-portal is not present."
-    test-snapd-desktop.check-dirs "$user_data" 2>stderr.log
+    session-tool -u test test-snapd-desktop.check-dirs /home/test/snap/test-snapd-desktop/current 2>stderr.log
     check_stderr stderr.log
 
     echo "The absence of the document portal service was recorded"
-    [ -f "$XDG_RUNTIME_DIR/.portals-unavailable" ]
+    test -f /run/user/12345/.portals-unavailable
```

Comment:
Is there a preference for `test` over square bracket op? Just curious.

Reply:
This is my personal preference:

I will use `[` in an if statement:

```
if [ "$stuff" = foo ]; then
   ...
fi
```
This looks like regular conditional statement in most languages.

I will use `test` when it acts as an assertion.
```
test "$stuff = foo
````
This looks like C `assert()`